lib/ui/widgets/post_content/widgets/link_preview/utils/preview_helper.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:html/parser.dart';
+import 'package:http/http.dart';
+import 'package:mooncake/ui/models/export.dart';
+
+Future<RichLinkPreview> fetchPreview(List<String> urls) async {
+  for (var i = urls.length - 1; i >= 0; i--) {
+    RichLinkPreview data = await fetchSinglePreview(urls[i]);
+    if (data != null) return data;
+  }
+  return null;
+}
+
+Future<RichLinkPreview> fetchSinglePreview(String url) async {
```

Comment:
Since this method is used only inside this file, I think we can make it private (prepending `_` to its name). What do you think?